### DISCUSSION TITLE
Wiki Frage: bei der Berechnung des Power Limits, die OpenDTU prüft wie viele Eingänge produzieren (>2W liefern) und skaliert entsprechend.

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#590] [Q&A]
Sorry falls ich nerve @schlimchen Bernhard, 

ich habe noch eine Frage fuer ein Thema dass auf die Wiki fehlt 🙈 

ich habe in Erinnerung, wenn die OpenDTU ein Limit an den Wechselrichter sendet, der Limit wird auf die MPPTs gleich verteilt.
z.B. Wechselrichter Limit von 600W auf einen HM-1500 (4 Eingaenge - 2 MPPT) wird als 300W pro MPPT definiert.

Ich habe vor ein paar Wochen in einer Diskussion gelesen, dass bei der Berechnung des Power Limits, die OpenDTU prüft wie viele Eingänge produzieren (>2W liefern) und skaliert entsprechend.

Letzteres wurde bedeuten, man koennte zwei Solar Paneele auf eine Seite (ein MPPT) der HM-1500 und die Batterie auf der anderen Seite (anderer MPPT):
Tagsueber wuerde der Dynamische Power Limiter korrekt funktionieren und das Limit auf die zwei Seiten teilen.
Genauso Nachts wuerde der Dynamic Power Limiter korrekt funktionieren und das Limit nur auf die Seite wo die Batterie angeschlossen ist, senden.

FRAGE: Stimmt das? "dass bei der Berechnung des Power Limits, die OpenDTU prüft wie viele Eingänge produzieren (>2W liefern) und skaliert entsprechend."
Und falls Ja, wie oft prüft die OpenDTU wie viele Eingänge produzieren (>2W liefern) und skaliert entsprechend? (wann wird die OpenDTU wissen, dass es Nacht ist und die Solar Paneele nicht mehr produzieren)

**Comment by spcqike** [comment]
die Überprüfung findest du in der Funktion ```setNewPowerLimit()``` im ```PowerLimiter```
https://github.com/helgeerbe/OpenDTU-OnBattery/blob/7946dfc0c2874c59225cf862d382843b84971810/src/PowerLimiter.cpp#L537-L547

Es wird also jedes mal, wenn das Limit gesetzt wird, geprüft.

Wie sich das Ganze über die verschiedenen Hoymiles hinweg verhält, also ob hoymilesintern 1 MPPT gleichbedeutend mit einem Eingang ist, oder nicht, kann ich dir nicht genau beantworten 😄

**Reply by Manos1966** [reply]
@spcqike ! Wo ist die Diskussion wo wir darueber gesprochen haben? Ich habe lange ohne Erfolg gesucht....

**Reply by spcqike** [reply]
die?
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/559#discussioncomment-7933683

**Reply by Manos1966** [reply]
Verdammt! Ich habe so lange danach gesucht. Danke 👍

**Comment by schlimmchen** [comment]
Dazu mag ich keinen Input liefern. Ich finde diese Funktion furchtbar und ich mag nicht dazu beisteuern, dass die auch noch im Wiki beschrieben wird.

**Reply by spcqike** [reply]
kann ich verstehen. ich kann mir auch nicht vorstellen, dass ein Mischbetrieb, wie er oben angesprochen wird, gut funktioniert.

Ich kann (glaube ich) grundsätzlich verstehen, warum es so umgesetzt wurde. aber elektrisch (Hardware) sauber ist es nicht. Warum sollte ich einen Eingang maximal und den zweiten gar nicht belasten wollen? Lieber beide halb belasten und dank geringere Ströme auch geringere Verluste und Belastung der Bauteile haben.

**Reply by Manos1966** [reply]
@schlimmchen OK, ich habe es mit dem Fall der gleichzeitiger Anbindung der Paneele und der Batterie etwas "ausgereizt" 😄 ich kenne schon deine Meinung zum Thema.

Es waere aber sinnvoll ein paar Erklaerungen in die Wiki zu haben, wie das **drosseln** des Wechselrichters funktioniert, den immer wieder tauchen die gleichen Fragen im Forum:

- Ohne Batterie: Wie wird das Limit geregelt wenn ich z.B. drei Paneele angeschlossen habe (anstatt vier)?

- Mit Batterie: Wie viele Eingaenge an die Batterie sollte ich anschliessen?

Ich habe in Wiki das Thema das Thema der Max Amperes pro Eingang (10A fuer HM, 15A fuer HMS) beschrieben und die Uebersetzung auf Watt abhaenging der 12V, 24V, 48V Batteriespannung.
Dies ist der eine Entscheidungsfaktor.

Ich persoenlich habe ich bis jetzt den Eindruck, ich muss alle MPPTs des Wechselrichters anschliessen, den der Befehl des Leistungslimits gleichmaessig auf die MPPTs geteilt wird.
Jetzt frage ich mich ob es ueberhaupt stimmt... und diese Erklaerung sollten wir (meiner Meinung nach) in die Wiki beschrieben haben.

**Reply by spcqike** [reply]
> Ich habe in Wiki das Thema das Thema der Max Amperes pro Eingang (10A fuer HM, 15A fuer HMS) beschrieben

verweis doch an dieser Stelle einfach aufs Datenblatt. Die Nutzer müssen schon gucken, dass alles elektrisch passt und sicher ist.

Die neuen HMS haben bspw ganz andere Werte. 
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/510f8573-2bff-447b-96b7-714ef28aa5f2)

Ich werd mir wohl im Laufe des Frühjahres einen HMS-2000-4T holen und 8 Trina Vertex S+ 435W anschließen. je 2 parallel pro Eingang. (die Module gabs für 70€/Stk). Bis ich dann mal ne Batterie habe, werde ich die "Verluste" (also das, was man nicht erzeugt, weil der WR drosselt) in Kauf nehmen. Später kommt dann ein entsprechender Laderegler und ne 48V Batterie.

Das Ganze sollte auch passen, da die Trina einen Kurzschlussstrom von "lediglich" 10.64A haben. also auch 2 parallel reißen die 25A nicht.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/67639815-4f31-4fbc-bc1c-fb85e7503de0)

Die Nennleistung in unseren Breitengraden liegt bei nur noch 332W pro Paneel, sodass ich ~2.6kW Nennleistung (bei ~3.5kWp) habe und damit 30% Überbelegung.

**Comment by Manos1966** [comment]
Hallo @spcqike 
ich habe ein bisschen gesucht und gefunden, Hoymiles verteilt ein Power-Limit **gleichmaessig** auf die MPPTs.
Das bedeutet, der HM-1500 (4 Eingänge, 2 MPPTs) wird ein Power-Limit von 1200W gleichmaessig auf die zwei MPPTs teilen, also   Eingang 1+2 600W und Eingang 3+4 600W.

Jetzt, wenn drei Eingaenge an die Batterie angeschlossen (1,2 und 3)  
50V Batterie, max A pro Eingang 10A
Eingang 1 (50Vx10A) 500W (max)
Eingang 2 (50Vx10A) 500W (max)
Eingang 3 (50Vx10A) 500W (max)

Powerlimit wird auf 1200W gesetzt und zum Hoymiles befehlt.
Hoymiles teilt 1200W auf zwei MPPTs also jeweils 600W
Eingang 1&2 zusammen 1000W fähig, liefern die 600W ohne Problem
Eingang 3 aber kann max 500W liefern, also 500W
Gesamt Ergebnis (600+500) 1100W (anstatt 1200W)

Daraufhin verstehe ich nicht ganz, zu welchem Zweck "die OpenDTU prüft wie viele Eingänge produzieren (>2W liefern) und **skaliert entsprechend**" wenn die Moeglichkeit nicht existiert zu erkennen, wieviel Max-Power pro MPPT verfuegbar ist.

Würde die OpenDTU wissen, dass MPPT A (Eingang 1&2) 1000W max liefern kann, und MPPT B (Eingang 3&4) 500W, dann koennte sie das Power-Limit von 1200W als 1400W-Befehl an die Hoymiles geben:
Hoymiles teilt 1400W auf zwei MPPTs also jeweils 700W
Eingang 1&2 zusammen 1000W fähig, liefern die 700W ohne Problem
Eingang 3 aber kann max 500W liefern, also 500W
Gesamt Ergebnis (700+500) 1200W!
Aber dies ist nicht was die OpenDTU macht, wenn ich deine Interpretation des Kodes richtig verstanden habe.


PS. Vielleicht ist aber die Funktion sinnvoll fuer die HMS Serie, sie hat jeweils ein MPPT pro Eingang 😉

**Reply by spcqike** [reply]
Die Funktion gibt es schon länger als die HMS Serie. 
Woher sie genau kommt, weiß ich nicht, ich denke aber sowas wie "ich habe einen HM-600, möchte aber nur 200W Einspeisen. ich könnte also 2 Paneele kaufen oder 2 Kabel an ne Batterie anschließen, oder nur eins. mit einem geht aber der einfache Limiter nicht richtig." 

ein Mischbetrieb oder nicht volle Belegung, halte ich aber für schwierig. Deine Beispiele gehen in den Zahlen gut auf, aber macht das technisch / elektrisch Sinn? zwei Eingänge "entspannen" bei 350W, der dritte leistet 500W und hat damit ne viel höhere Belastung und Verschleiß? 

Auch solche Szenarien "PV Paneele an Eingang 1+2, Batterie an Eingang 3+4" ... wozu? die Batterie muss auch irgendwie geladen werden, braucht also eigene Paneele. Warum sollte ich die Paneele an Eingang 1+2 drosseln und die dortigen Überschüsse NICHT in die Batterie laden? So, wie es 80% der onBattery Benutzer machen, also mit einem oder mehreren Solarladereglern, einer Batterie und Solar-Passtrough, macht es doch mehr Sinn.

Wie gesagt, ich verstehe die Funktion der Dynamik teilweise, aber auch nicht ganz. Ich hab auch nicht dazu beigetragen, dass sie rein kommt. Ich wollte diese Stelle in der Vergangenheit mal erweitern, da ich bisher nur den HM-600 mit 2 unterschiedlich verschatteten Modulen habe. Die Erweiterung kam aber nicht rein. 

(ich hatte die Dynamik so gestalltet, dass nicht nur geschaut wurde, ob der Eingang >2W produziert, sondern ebenfalls, ob mindestens ein Eingang am Drossellimit "hängt", mindestens ein weiterer aber eben nicht. und dann den Limiter entsprechend höher gestellt. bspw. bei meinem HM-600 und sagen wir 200W, 100W pro MPPT. einer liefert 100W, der zweite nur 50W. in meinem Fork ging das Limit dann eben auf 300W hoch, damit der erste Eingang hoffentlich 150W liefert und der zweite weiterhin seine 50W...)


mal gucken was Hoymiles macht. Deye's neue G4 Wechselrichter haben ja scheinbar eine AC-seitige Drosselung. Damit wären all solche Überlegungen hinfällig. allerdings erscheint mir die Regelzeit bisher recht hoch, bis sich das System eingeschwungen hat. Vielleicht zieht Hoymiles ja per Software oder mit neuer Hardware nach.

**Comment by Manos1966** [comment]
Ich muss schnell betonen (bevor schlimchen 'was schreibt... 🤣) : Ich schliesse immer ALLE Eingaenge & homogen (gleiche Quelle) an! 😉 

Die Diskussion hier dient **ausschließlich zum** (meinem) **Verständnis**, weil wir etwas diesbezüglich in die Wiki schreiben sollten.
Ich "muesste" mein Beispiel auf 1200W Limit hoch-schrauben um das Problem erkenntlich zu machen. Das offizielle Einspeise-Limit sind 600W (nicht Mal 800W), bei diesem Limit taucht das Problem gar nicht auf.  
Viele OpenDTUonBattery "Einsteiger" stellen diese Frage, Verständlicherweise.  

Ausserdem, (ich "betreue" fuenf Nachbarn) ich habe:
- einen Nachbar, der hat drei Paneelle ansgeschlossen, weil er keinen Platz fuer das vierte hatte
- einen der nur zwei auf die 4-Eingaenge Hoymiles hat. Beim letzeren habe ich mich selber die Frage gestellt (vor einem Jahr) ob wir sie auf der gleichen Seite des Wechselrichters anschliessen sollten, oder jeweils auf einer Seite
- einen der jeweils zwei mit unterschiedlicher Ausrichtung auf die 4-Eingaenge Hoymiles hat.  

Waere eine Nulleinspeisung bei einen der oben erwähnten Fälle erwünscht, haette man wahrscheinlich gewundert, warum sie nicht 100% prozentig funktioniert. 
Damals war die OpenDTUonBattery Nulleispeisung gar kein Thema. 
Auch heute, ist Nulleinspeisung kein Thema 😉 Nulleinspeisung ist interessant wenn man Batterien hat.
Aber zu Wissen, wie etwas funktioniert, macht das Leben Interessant...


Neben-Information (unwichtig, aber informativ): Anscheinend gibt es nur zwei Wechselrichter heute die "intelligent" das Power Limit zwischen den MPPTs aufteilen.
Hier ist das Video:
[die besten Wechselrichter zum Drosseln!](https://youtu.be/9W1WTIYNEyo?t=614)

**Reply by spcqike** [reply]
das Video hatte ich auch im Hinterkopf. allerdings sind es nicht "zwei" Wechselrichter, sondern alle Deye Mikrowechselrichter der 4. Generation. also von 300,400,500W (1-phasig, 1MPPT, 1 Eingang), über 600W, 800W, 1000W (1-phasig, 2 MPPT, 2 Eingänge) bis zu 1300W, 1600W, 1800W, 2000W, 2200W (1-phasig, 4MPPT, 4 Eingänge)

Die Regelgeschwindigkeit, die er in dem Video zeigt, ist aber unterirdisch. grade das Beispiel, wo das zweite NT wieder zugeschalten wird und der WR, trotz 400W Limit, erstmal > 1 Minute 600W einspeist.
Auch das Hochregeln/Anfahren. ja, Wechselrichter dürfen nicht zu schnell regeln, aber das, was er da zeigt, ist total langsam. die Hoymiles regeln weit schneller. mein HM-600 geht binnen Sekunden (1-2s, je nach Übertragung) von 600W runter auf 50W und auch wieder hoch, wenn ich es denn will. 

Nachtrag:
bei den 1MPPT Wechselrichtern macht es halt keinen Sinn :D aber wahrscheinlich liegt dort in der Software der gleiche Regelalgorithmus. also ne ähnliche Trägheit.

**Reply by spcqike** [reply]
@Manos1966 auch wenns hier nicht ganz passt, aber die Regelgeschwindigkeit lässt mir grad keine Ruhe 😄 

du hast einen HM-1500, oder? Kannst du sagen, wie schnell der hoch und runter regelt? 
scheinbar darf ein Wechselrichter in Deutschland nicht schneller als 400W/s regeln, zumindest finde ich das hier und da im Netz (bspw. [hier](https://www.akkudoktor.net/forum/stell-dein-batterie-powerwall-projekt-vor/regelverzoegerung-beim-victron-multiplus/paged/3/)). die VDE, die es klar regeln soll, liegt mir nicht offen vor. 

Wie anderswo geschrieben, plane ich einen HMS-2000 zu holen. Wenn die nun wirklich so langsam regeln, wäre es ja "besser" einfach 2 HM-1500 zu nehmen. dann käme man auf effektiv 800W/s dank zwei Wechselrichtern..... oder?

**Reply by Manos1966** [reply]
😄 @spcqike 

Testen kann ich. 
Definiere die Parameter. Ich meine, wie willst du, dass ich die Geschwindigkeit messen soll? 

400W/s ist meiner Meinung nach nicht schlecht, binnen 5 Sekunden ist er von 0W auf 2000W gegangen.

Der HMS2000-4T hat ein CMT2300A Kommunikationseinheit. 
Seine Steuerung sollte kein Thema sein...
Meine Empfehlung dazu: Nur eine Phase 👎  Warum nicht lieber einen drei-phasigen Wechselrichter? Die Last wird dadurch auf drei Leitungen geteilt, anstatt alles nur auf eine.


Mehr Sorge bereiten mich die neuen Wechselrichter, die mit WiFi gesteuert werden. Anscheinend akzeptieren sie Befehle nur ein Mal pro Minute (und das klingt sogar gut, die ersten Berichte sprachen von 5 Minuten.... 🙈 )

**Reply by Manos1966** [reply]
Uebrigens, die OPenDTU ist kein "Real-time" System. 
Eine gewisse Zeitverzoegerung existert zwischen:
- Lesen der Stromzaehler-Daten 
(allein TASMOTA hat 30Sec Standard Daten Übertragung. Man kann die Uebertragung auf 10Sek runter setzten... AUSSER man stellt gewisse Daten auf Real-Time um 🤣 Ich weiss aber nicht wie viele Anwender wissen das...)
- Neues Limit an Wechselrichter senden

Daher, die paar Sekunden um das neue Limit zu erreichen, das sollte kein Thema sein.

**Reply by spcqike** [reply]
Ja klar besteht eine gewisse Verzögerung immer. 

Ja Tasmota sendet mir zu selten. Daher nutze ich ja auch JSON als Input. Da zieht die openDTU-ob die neusten Werte ja kurz bevor sie benötigt werden. Da sollte der Wert vom Stromzähler halt <1s alt sein. 

Ich weiß nicht ob und wie du das messen kannst. Wenn du von 1500W auf 100W gehst. Und wieder hoch. Wie lange es tatsächlich dauert bis der WR entsprechend läuft. Mit der DTU alleine wird es wahrscheinlich nicht klappen. 


wozu einen 3 phasigen WR? Die sind weit teurer, oder? Man darf immerhin … 3.6kVA pro Phase einspeisen, bevor man einen echten 3-phasigen wechselrichter braucht. (Drei 1-phasige inverter die kommunizieren, wie die multiplus, gehen auch) es ist also technisch möglich und erlaubt jeweils einen 1500er und 2000er auf jeder Phase zu betreiben.

**Reply by Manos1966** [reply]
"Man darf immerhin … 3.6kVA pro Phase einspeisen,"
mit (viel) Vorsicht zu genießen. Wenn der Wasserkocher für 30Sek 3kW verbraucht, ist dies kein Problem.
Wenn der Ladecharger fuer das Elektroauto mit 3kW die ganze Nacht das Auto lädt ich wurde kein Auge zu machen.... 🙈 

Am Ende des Tages, es haengt von der Qualitaet der Stromleitungen an.

> Als Faustregel für den Kabelquerschnitt der Hausinstallation können Sie sich merken: 
1,5 mm² für Beleuchtung und weniger belastete Steckdosen. 
2,5 mm² für hoch belastete Steckdosen (Wäschetrockner, Geschirrspüler, Heizkessel, etc.) 
4 mm² für Elektroherde und stärkere Maschinen in der Hobbywerkstatt oder Garage.

Kurz gesagt, ich wurde keine 3,6kW in einer Steckdose einspeisen ohne den Kabelquerschnitt zu wissen.
![Untitled](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/034295f4-2b87-47cd-935f-daefc4cfc955)

**Reply by spcqike** [reply]
Selbst mit bekanntem kabelquerschnitt würde ich das nicht machen. Immerhin sind SchuKo Dosen und Stecker nicht für 16A Dauerlast ausgelegt. 
Aber ich hab auch nie gesagt, dass man das pro Steckdose machen sollte. Nur, dass man es pro **Phase** darf. Die eigene Hausinstallation / Leitungen sollte natürlich jeder im Hinterkopf behalten. 

Es ging mir halt um deine Aussage 

> Meine Empfehlung dazu: Nur eine Phase 👎 Warum nicht lieber einen drei-phasigen Wechselrichter? Die Last wird dadurch auf drei Leitungen geteilt, anstatt alles nur auf eine.

Der HMT sind teurer als die HM/HMS. Und die wenigsten Leute werden einen HMT brauchen, da sie bisher wahrscheinlich noch keine 3.6kVA auf allen drei Phasen haben. 

Am Ende saldieren die Stromzähler in Deutschland ja auch. 

Die Leistung muss auch nicht auf allen drei Phasen verteilt werden. 2kW auf einer Phase für 24/7 sind halt unkritisch. Sonst dürfte man auch keine 1-phasigen bautrockner / heizer oder Durchlauferhitzer betreiben.

**Comment by Manos1966** [comment]
> Ja Tasmota sendet mir zu selten.

Bevor ich es vergesse:

Wie oft die Werte übermittelt werden, regelt das Tasmota-KonsolenKommando "telePeriod" als Grundeinstellung und dazu strom/spannungs/leistungsabhängig diverse (u.a. "powerdelta"), wann zusätzlich bei Änderungen der Werte (mit prozentual einstellbaren Schwellen).

Tasmota command 'TelePeriod' See current value and force publish STATE message in MQTT
0 = disable telemetry messages
1 = reset telemetry period to firmware default (TELE_PERIOD)
10..3600 = set telemetry period in seconds (default = 300)
Also 10Sec ist das min Teleperiod...

**ABER fuer Hichi gibt es eine andere Moeglichkeit...**

Aus dem Tasmota Wiki:

1,77070100100700ff@1,P,W,P,**18**  
the last entry is the number of digits + **an optional flag to send the value as fast as possible code = 16** 
so 16+2 = 18 means 2 digits + fast send

Beispiel:
Wenn deine Zeile so aussieht:  1,77070100100700ff@1,P,W,P,**0** 
rechnest du 0+16=16 und änderst es so: 1,77070100100700ff@1,P,W,P,**16**
Also immer die letzte Zahlt in Descriptor plus oder minus 16.

Alles hier Beschrieben: https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-metrics
